### PR TITLE
test: Make summary row bold-red if any test failed and show failed tests at end of table

### PR DESCRIPTION
**Pull request by laanwj**
Make the summary row of the test runner bold red if *any* test fails. This helps visibility if something fails.
(yesteryday I had a snafu where I missed that `feature_blocksdir.py` had failed because it's one of the earlier tests in the list, this intends to avoid that in the future)

Before:
testfailold

After:
testfailnew

If tests pass it still looks the same:

testok

### CODE REVIEW

**Comment by laanwj** [comment]
Added a commit (by @ajtowns) that moves the failed tests to the bottom of the list, to avoid needing to scroll to find out what failed.

**Comment by MarcoFalke** [comment]
utACK d2c6a371a036941bf9c4d58a612ced0970b9b056

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -455,6 +459,14 @@ def __init__(self, name, status, time):
         self.time = time
         self.padding = 0
 
+    def sort_key(self):
+        if self.status == "Passed":
+            return 1, self.name.lower()
+        elif self.status == "Failed":
+            return 2, self.name.lower()
+        elif self.status == "Skipped":
+            return 0, self.name.lower()
```
Comment (jnewbery) [inline test/functional/test_runner.py]: My preference would be to place skipped tests below passed tests for visibility, so the order would be:

<passed tests>
<skipped tests>
<failed tests>
SUMMARY

**Review by jnewbery — COMMENTED** [review]
Tested ACK d2c6a371a036941bf9c4d58a612ced0970b9b056 with one minor suggested change

File: test/functional/test_runner.py (hunk 1 above)
Comment (laanwj) [reply test/functional/test_runner.py]: Fair point, I'll swap the passed and skipped

**Comment by laanwj** [comment]
Swapped sort order of skipped and passed tests per @jnewbery 
d2c6a371a036941bf9c4d58a612ced0970b9b056 → f92541f7ea66b81343426db0b40f7ba4062df342

**Comment by jnewbery** [comment]
Tested ACK ffb033a6d5d846b19a2b595f7d522d5b73cfca6f

**Comment by playtuff42** [comment]
Can't seam to figure out where or how to get onto my comp to see pull requests put in my own pull. Could use some help user was not the intended recipient of file.  I need to know who I'm helping and then  I will be able to Get some more information....

**Pull request merged**